### PR TITLE
Feature/Add loading behind image load and improve UX

### PR DESCRIPTION
**Pull request by romeoonisim**
### Description
- improved the  UX of the web app
- added image placeholder until image loaded

### CODE REVIEW

File: app/components/img-fancy-load.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import Component from '@ember/component';
+
+export default Component.extend({
```
Comment (biancamariacampean) [inline app/components/img-fancy-load.js]: Why do you not use class for components?

File: app/components/img-fancy-load.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	},
+
+	_initImage() {
+		const img = new Image();
+
+		img.src = this.get('src');
+
+		img.addEventListener('load', () => {
+			// this is to handle if the component has been destroyed before finishing in tests
+			if (this.isDestroyed) {
```
Comment (alexandru-calinoiu) [inline app/components/img-fancy-load.js]: Is a little strange that we need this code in only for tests, ideally tests should not require us to write extra code.

It also makes me think that this is a potential memory leak.

Any other ways we could do this? Maybe remove the listeners when the component destroys?

Comment (romeoonisim) [reply app/components/img-fancy-load.js]: It had to do with something @biancamariacampean suggested. I've refactored it

**Review by biancamariacampean — COMMENTED** [review]
You should also check if the image is broken
image

**Comment by romeoonisim** [comment]
> You should also check if the image is broken
> image

I'm already making the fix for this in another PR.

**Pull request merged**